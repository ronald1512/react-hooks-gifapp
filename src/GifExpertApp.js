import React, { useState } from 'react'
import { CategoryAdd } from './components/CategoryAdd'
import GifGrid from './components/GifGrid'

//Snippet: rafc
export const GifExpertApp = ({defaultCategory=[]}) => {
    //const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
    /**Si quiero agregar categories dinámicamente no puedo hacerlo de esta forma porque:
     * Primero está declarado como constante
     * Segundo, ¿Cómo le diria a react que hubo un cambio en categories y que necesito que vuelva a renderizar?
     * 
     * Entonces, debemos usar el hook de useState, ya que necesitamos que categories cambie dinamicamente.
    */

    //const [categories, setcategories] = useState(['One Punch'])
    const [categories, setcategories] = useState(defaultCategory)

    // const handleAdd = ()=>{
    //     setcategories([...categories,'HunterXHunter']); //operador spreed!!
    //     //O tambien se puede hacer asi: setCategories( cats => [...cats, 'HunterXHunter']);
    // }

    return (
        <>
            <h2>GifExpertApp</h2> 
            <CategoryAdd setcategories={setcategories}/>
            <hr />
            {/* <button onClick={handleAdd}>Agregar</button> */}
            <ol>
                {
                    categories.map(category=> (
                        <GifGrid 
                            key={category}
                            category={category}/>
                    ))
                }
            </ol>
        </>
    )
}
