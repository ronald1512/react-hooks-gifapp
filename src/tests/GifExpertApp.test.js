import { shallow } from "enzyme"
import { GifExpertApp } from "../GifExpertApp"



describe('Probando GifExpertApp', () => {
    

    test('debe de mostrarse correctamente', () => {
        
        const wrapper = shallow(<GifExpertApp/>);

        expect(wrapper).toMatchSnapshot();
    });



    test('debe de mostrar una lista de categorÃ­as', () => {
        const categories=['One Punch', 'Dragon Ball'];

        const wrapper = shallow(<GifExpertApp defaultCategory={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    


    
})
