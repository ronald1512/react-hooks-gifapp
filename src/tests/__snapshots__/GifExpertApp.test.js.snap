// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando GifExpertApp debe de mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setcategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Probando GifExpertApp debe de mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setcategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
